lib/queue.js

```diff
@@ -621,6 +626,9 @@ Queue.prototype.add = function(name, data, opts){
     data = name;
     name = Job.DEFAULT_JOB_NAME;
   }
+  if (!opts) opts = {};
```

Comment:
this should not be needed since we already take care of it here: https://github.com/mxstbr/bull/blob/81d7540ac663ff4dcab30a9807067bd7d58105cb/lib/queue.js#L96

Reply:
This is in `Queue.prototype.add`, this is not the main queue options. `opts` is undefined here quite often since the tests (and users too probably) do `Queue.add(data)` without specifying any special options.

Reply:
ah ok, sorry, I did not see it right due to github removing all the lines between the queue constructor and this line.
But please change the line to:
```
opts = _.cloneDeep(opts || {});
```
to avoid sideeffects.
Also in line 632 we do not need to check for opts anymore.

Reply:
No worries, I know how it goes :blush: Will clean that up, one second!

Reply:
Done, good catch